### PR TITLE
fix(): #9009 artifacts

### PR DESCRIPTION
**Pull request by ShaMan123**
## Motivation

> I want this removed because I think in the future it is a potential bug
Once we remove type getter this thing will become an artifact

_Originally posted by @ShaMan123 in https://github.com/fabricjs/fabric.js/pull/9009#discussion_r1227187374_
            

## Description

I did not imagine that would happen so fast.

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
Fixes `textbox.type` and removes future bug potential with the filters.

**Comment by asturur** [comment]
mmmmm
those get type are there for backaward compat?
What is the bug actually?

**Comment by ShaMan123** [comment]
The bug is textbox.type==='i-text'
Filters have a grtter from super class, enogh just to set static type

**Comment by asturur** [comment]
in fabric6 instances need to have a type property that return the fabric 5 style type.
We stopped using internally and we will remove if forward in time.
That `i-text' is on purpose.

File: src/filters/ColorMatrixFilters.ts [hunk 1]
```diff
@@ -3,10 +3,6 @@ import { classRegistry } from '../ClassRegistry';
 
 export function createColorMatrixFilter(key: string, matrix: number[]) {
   const newClass = class extends ColorMatrix {
-    get type() {
-      return key;
-    }
```
Comment (asturur) [inline src/filters/ColorMatrixFilters.ts:8]: if this still works because of BaseFilter, this we can remove

File: src/shapes/IText/IText.ts [hunk 2]
```diff
@@ -204,10 +204,6 @@ export class IText<
 
   static type = 'IText';
 
-  get type() {
-    return 'i-text';
-  }
```
Comment (asturur) [inline src/shapes/IText/IText.ts:209]: this should stay.

Comment (ShaMan123) [reply src/shapes/IText/IText.ts:209]: But is breaks subclassing
Textbox is broken

Comment (ShaMan123) [reply src/shapes/IText/IText.ts:209]: We should instead override isType like you did with object

Comment (ShaMan123) [reply src/shapes/IText/IText.ts:209]: Fixed like you did with object

File: .codesandbox/templates/vanilla/src/index.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import * as fabric from 'fabric';
 import './styles.css';
 
 const el = document.getElementById('canvas');
-const canvas = new fabric.Canvas(el);
+const canvas = (window.canvas = new fabric.Canvas(el));
```
Comment (ShaMan123) [inline .codesandbox/templates/vanilla/src/index.ts:5]: for ease of dev

**Review by ShaMan123 — COMMENTED** [review]
This is ready and satisfies all requirements

**Review by asturur — APPROVED** [review]

**Review by asturur — APPROVED** [review]

**Pull request merged**